src/apiQueries/useReceiversReceiverId.ts

```diff
@@ -14,22 +14,19 @@ export const useReceiversReceiverId = <T>({
   dataFormat: "receiver" | "paymentReceiver";
   receiverWalletId?: string;
 }) => {
-  const query = useQuery<ApiReceiver, AppError>({
+  const query = useQuery<ReceiverDetails | PaymentDetailsReceiver, AppError>({
     queryKey: ["receivers", dataFormat, receiverId, { receiverWalletId }],
     queryFn: async () => {
-      return await fetchApi(`${API_URL}/receivers/${receiverId}`);
+      const response = await fetchApi(`${API_URL}/receivers/${receiverId}`);
+      return dataFormat === "receiver"
+        ? formatReceiver(response)
+        : formatPaymentReceiver(response, receiverWalletId);
```

Comment:
It might be safer to check if there is `receiverWalletId` because we need it in the `formatPaymentReceiver()`. The data format might be correct, but `receiverWalletId` might be missing. Or we can check for both to be 100% sure.

```suggestion
      return receiverWalletId
        ? formatReceiver(response)
        : formatPaymentReceiver(response, receiverWalletId);
```